webpack/JobWizard/steps/Schedule/RepeatOn.js

```diff
@@ -1,54 +1,212 @@
 import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import { TextInput, Grid, GridItem, FormGroup } from '@patternfly/react-core';
-import { translate as __ } from 'foremanReact/common/I18n';
+import {
+  TextInput,
+  Grid,
+  GridItem,
+  FormGroup,
+  TimePicker,
+  Checkbox,
+  Select,
+  SelectOption,
+  SelectVariant,
+} from '@patternfly/react-core';
+import { range } from 'lodash';
+import { translate as __, documentLocale } from 'foremanReact/common/I18n';
 import { SelectField } from '../form/SelectField';
 import { repeatTypes } from '../../JobWizardConstants';
+import { helpLabel } from '../form/FormHelpers';
+
+const getWeekDays = () => {
+  const locale = documentLocale().replace(/-/g, '_');
+  const baseDate = new Date(Date.UTC(2017, 0, 2)); // just a Monday
+  const weekDays = [];
+  for (let i = 0; i < 7; i++) {
+    try {
+      weekDays.push(baseDate.toLocaleDateString(locale, { weekday: 'short' }));
+    } catch {
+      weekDays.push(baseDate.toLocaleDateString('en', { weekday: 'short' }));
+    }
+    baseDate.setDate(baseDate.getDate() + 1);
+  }
+  return weekDays;
+};
 
 export const RepeatOn = ({
   repeatType,
   setRepeatType,
   repeatAmount,
   setRepeatAmount,
+  repeatData,
+  setRepeatData,
 }) => {
   const [repeatValidated, setRepeatValidated] = useState('default');
+  const [minuteOpen, setMinuteOpen] = useState(false);
   const handleRepeatInputChange = newValue => {
     setRepeatValidated(newValue >= 1 ? 'default' : 'error');
     setRepeatAmount(newValue);
   };
+
+  const days = getWeekDays();
+  const handleChangeDays = (checked, event) => {
+    const {
+      target: { name },
+    } = event;
+    setRepeatData({
+      daysOfWeek: { ...repeatData.daysOfWeek, [name]: checked },
+    });
+  };
+  const atComponent = () => (
+    <FormGroup label={__('At')}>
+      <TimePicker
+        aria-label="repeat-at"
+        className="time-picker"
+        time={repeatData.at}
+        placeholder="hh:mm"
+        onChange={newTime => {
+          setRepeatData({ ...repeatData, at: newTime });
+        }}
+        is24Hour
+        invalidFormatErrorMessage={__('Invalid time format')}
+      />
+    </FormGroup>
+  );
+  const getRepeatComponent = () => {
```

Comment:
This section takes more than 100 lines,
can you move it into a separate file, e.g `./Schedule/Repeat{Component/Type}.js`